### PR TITLE
RDKDEV-346: RDKServices startup-order support

### PR DESCRIPTION
**Pull request by arun-madhavan-013**
Reason for change: Provision to add startup-order
support for plugins from build recipe. Require: https://github.com/rdkcentral/Thunder/pull/1071
Test Procedure: Please see ticket for details.
Risks: Medium
Source: Comcast
License: Inherited
Upstream-Status: Pending

Signed-off-by: Arun Madhavan <arun_madhavan@comcast.com>
--------------------------------------------
How To Use:
Pass the required build configuration for startup-order through build flags as below.
`
EXTRA_OECMAKE += " \
                  -DPLUGIN_DEVICEIDENTIFICATION_STARTUPORDER=30 \
                  -DPLUGIN_DEVICEINFO_STARTUPORDER=30 \
                  -DPLUGIN_DISPLAYINFO_STARTUPORDER=45 \
                  -DPLUGIN_LOCATIONSYNC_STARTUPORDER=30 \
                  -DPLUGIN_MESSENGER_STARTUPORDER=30 \
                  -DPLUGIN_MONITOR_STARTUPORDER=30 \
                  -DPLUGIN_PLAYERINFO_STARTUPORDER=35 \
                  -DPLUGIN_RDKSHELL_STARTUPORDER=25 \
                  -DPLUGIN_TEXTTOSPEECH_STARTUPORDER=40 \
                 "
`
Eg:
> cat recipes-extended/rdkservices/rdkservices_git.bbappend
EXTRA_OECMAKE += " \
                  -DPLUGIN_DEVICEIDENTIFICATION_STARTUPORDER=30 \
                  -DPLUGIN_DEVICEINFO_STARTUPORDER=30 \
                  -DPLUGIN_DISPLAYINFO_STARTUPORDER=45 \
                  -DPLUGIN_LOCATIONSYNC_STARTUPORDER=30 \
                  -DPLUGIN_MESSENGER_STARTUPORDER=30 \
                  -DPLUGIN_MONITOR_STARTUPORDER=30 \
                  -DPLUGIN_PLAYERINFO_STARTUPORDER=35 \
                  -DPLUGIN_RDKSHELL_STARTUPORDER=25 \
                  -DPLUGIN_TEXTTOSPEECH_STARTUPORDER=40 \
                 "

### CODE REVIEW

**Review by anand-ky — COMMENTED** [review]
Changes look good. There are many patches related to startup order (Mostly in XClass) that would need to be updated to accommodate these changes. Can we pls check and see if these can come together.

**Comment by arun-madhavan-013** [comment]
Thank you @anand-ky ,

Could you please point me to any reference there so that I can check what all can be made generic.

Regards
Arun

**Comment by anand-ky** [comment]
Pls refer to https://ccp.sys.comcast.net/browse/RDKTV-177

**Comment by arun-madhavan-013** [comment]
Will re-open with related changes and PR links.

**Comment by arun-madhavan-013** [comment]
Linking related PR: https://github.com/rdkcentral/Thunder/pull/1071

**Comment by anand-ky** [comment]
@emutavchi - Pls review changes in WebKitBrowser

File: AVInput/AVInput.config [hunk 1]
```diff
@@ -1,3 +1,9 @@
 set (autostart false)
 set (preconditions Platform)
 set (callsign org.rdk.AVInput)
+
+if(PLUGIN_AVINPUT_STARTUPORDER)
+map()
```
Comment (emutavchi) [inline AVInput/AVInput.config:6]: Why do we need map() here? isn't it enough to just set env variable? like autostart above

Comment (arun-madhavan-013) [reply AVInput/AVInput.config:6]: The reference patch had similar configuration; didn't wanted to deviate much from it.
But it seems not required.

Comment (emutavchi) [reply AVInput/AVInput.config:6]: I think it'd be more readable if you drop not needed map()/end().

Comment (arun-madhavan-013) [reply AVInput/AVInput.config]: Updated the changes after a local build verification. Will put the same for sprint verification also.

**Review by emutavchi — COMMENTED** [review]
Looks good for browser callsigns we use. @MFransen69  could you please review as well?

**Comment by MFransen69** [comment]
> Looks good for browser callsigns we use. @MFransen69 could you please review as well?

Hi Eugene, LGTM :)

**Review by anand-ky — APPROVED** [review]

**Pull request merged**